dev/com.ibm.ws.transport.http/src/io/openliberty/http/netty/channel/utils/HeaderValidator.java

```diff
@@ -0,0 +1,177 @@
+    */
+    public enum FieldType{NAME, VALUE}
+
+    private HeaderValidator() {
+        //Utility Singleton
+    }
+
+
+    /**
+     * Peforms processing of a header field (name or value).
+     * 
+     * This method normalizes a header field:
+     *      Ensures a non-null input if field type is {@link FieldType#NAME}.
+     *      Trims the input if it is non-null.
+     *      Substitutes a null input with an empty string if the field type is
+     *          {@link FieldType#VALUE}.
+     *      Lowercases the token if the field type is {@link FieldType#NAME}
+     * 
+     * @param token a raw header field token; may be {@code null} for values, 
+     *      but not for names.
+     * @param type whether this token is a header name or value
+     * @param config the HTTP configuration object
+     * @return processed and possibly normalized header field, ensuring to 
+     *      comply with the configured validation requirements
+     * @throws IllegalArgumentException if the field is invalid (too long, 
+     *      contains illegal characters, or null name)
+     */
+    public static String process(String token, FieldType type, HttpChannelConfig config){
+
+        if(type == FieldType.NAME && token == null){
+            throw new IllegalArgumentException("Header name must not be null");
+        }
+        String normalized = (token == null) ? "": token.trim();
+
+        if(type == FieldType.NAME){
+            normalized = normalized.toLowerCase();
+        }
+        
+        return validate(normalized, type, config);
+
+    }
+
+    /**
+     * Validates and cleans a header field:
+     *  Enforces the maximum allowed length as specified by {@link HttpChannelConfig}
+     *  Returns token as-is if validation is disabled.
+     *  Checks that a header name token is complaint with the RFC "tchar" pattern.
+     *  Disallows trailing CR or LF characters.
+     *  Ensures that an inline CR is followed by LF, and LF is followed by space or tab.
+     *  Replaces valid CR or LF (folding sequence) with space character.
+     *  For character outside of the printable ASCII range, if the masked code point 
+     *      is CR or LF, it is replaced with '?'.
+     * @param token the header field test after it has been normalized
+     * @param type  whether this token is a header name or a header value
+     * @param config the HTTP configuration object
+     * @return the validated (and clean) header field
+     * @throws IllegalArgumentException if the field fails one of the checks
+     */
+    private static String validate(String token, FieldType type, HttpChannelConfig config){
+
+        final int MAX_FIELD_SIZE = config.getLimitOfFieldSize();
+
+        // Check for length limit
+        if (token.length() > MAX_FIELD_SIZE) {
+            throw new IllegalArgumentException(token +
+                                               " exceeds the maximum allowed length of " + MAX_FIELD_SIZE + " characters");
```

Comment:
Should this exception message be translated? Does the end use see it?

Reply:
I won't comment on all of the exception here but the same question for all of them.